### ISSUE TITLE
Visiting / after fresh install results in a 500

### ISSUE THREAD

**Issue by botandrose** [BartMassey/rocket-pastebin#1]
```
'<unnamed>' panicked at 'could not open upload directory: Error { repr: Os { code: 2, message: "No such file or directory" } }', /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libcore/result.rs:845
```

Looks to me like an empty upload directory should be checked into the repo, or the server should mkdir it if it doesn't exist.

**Comment by BartMassey** [comment]
Yes, known issue I keep forgetting to fix. Thanks for the report!

It's not clear which is the right fix. Old versions of Git won't allow checkin of empty directories. On the other hand, the server needs to be started where it can find the templates, so having it dynamically create the upload directory seems silly.

I'll do the first thing for now, since it's easier.

**Comment by botandrose** [comment]
Yeah, you probably know this, but the normal workaround is to check in an empty `.keep` file in the directory, thus making it not empty.

**Comment by BartMassey** [comment]
Yeah, but then you have to make sure that path (`upload/.keep`) is never reachable by the app, or bad things could happen. I plan to add support for user-named paste files soon. I'll probably just mkdir as needed, actually, but check to make sure that the `templates/` directory exists first to avoid polluting the namespace. Friggin' Git.

**Comment by BartMassey** [comment]
Fixed in commit 2cf2713. Decided checking the parent directory first was a bad idea: who knows how this'll be installed?

**Comment by botandrose** [comment]
Confirmed.
